sdk/core/core-https/src/nodeHttpsClient.ts

```diff
@@ -0,0 +1,274 @@
+        host: proxySettings.host,
+        port: proxySettings.port,
+        headers: request.headers.raw()
+      };
+      if (proxySettings.username && proxySettings.password) {
+        proxyAgentOptions.auth = `${proxySettings.username}:${proxySettings.password}`;
+      }
+      proxyAgent = new httpsProxyAgent(proxyAgentOptions);
+    }
+    return proxyAgent;
+  } else if (request.keepAlive) {
+    if (!keepAliveAgent) {
+      keepAliveAgent = new https.Agent({
+        keepAlive: true
+      });
+    }
+
+    return keepAliveAgent;
+  } else {
+    return https.globalAgent;
+  }
+}
+
+function isReadableStream(body: any): body is NodeJS.ReadableStream {
+  return body && typeof body.pipe === "function";
+}
+
+class ReportTransform extends Transform {
+  private loadedBytes = 0;
+  private progressCallback: (progress: TransferProgressEvent) => void;
+  _transform(chunk: string | Buffer, _encoding: string, callback: Function): void {
+    this.push(chunk);
+    this.loadedBytes += chunk.length;
+    this.progressCallback({ loadedBytes: this.loadedBytes });
+    callback();
+  }
+
+  constructor(progressCallback: (progress: TransferProgressEvent) => void) {
+    super();
+    this.progressCallback = progressCallback;
+  }
+}
+
+/**
+ * A HttpsClient implementation that uses Node's "https" module to send HTTPS requests.
+ */
+export class NodeHttpsClient implements HttpsClient {
+  public async sendRequest(request: PipelineRequest): Promise<PipelineResponse> {
+    const abortController = new AbortController();
+    let abortListener: ((event: any) => void) | undefined;
+    if (request.abortSignal) {
+      if (request.abortSignal.aborted) {
+        throw new AbortError("The operation was aborted.");
+      }
+
+      abortListener = (event: Event) => {
+        if (event.type === "abort") {
+          abortController.abort();
+        }
+      };
+      request.abortSignal.addEventListener("abort", abortListener);
+    }
+
+    if (request.timeout > 0) {
+      setTimeout(() => {
+        abortController.abort();
+      }, request.timeout);
+    }
+
+    if (request.formData) {
+      const formData = request.formData;
+      const requestForm = new FormData();
+      for (const formKey of Object.keys(formData)) {
+        const formValue = formData[formKey];
+        if (Array.isArray(formValue)) {
+          for (const subValue of formValue) {
+            requestForm.append(formKey, subValue);
+          }
+        } else {
+          requestForm.append(formKey, formValue);
+        }
+      }
+
+      request.body = requestForm;
+      request.formData = undefined;
+      const contentType = request.headers.get("Content-Type");
+      if (contentType && contentType.indexOf("multipart/form-data") !== -1) {
+        request.headers.set(
+          "Content-Type",
+          `multipart/form-data; boundary=${requestForm.getBoundary()}`
+        );
+      }
+    }
+
+    if (!request.skipDecompressResponse) {
```

Comment:
Do we want to clear `Accept-Encoding` header for the `else` case?

Reply:
I suppose we could... but what would the scenario be? The user sets it as a manual header in the operation or another policy in the pipeline sets it?